### DISCUSSION TITLE
How to get resource URL of uploaded file?

### DISCUSSION THREAD

**Discussion by jsefiani** [aws/aws-sdk-js-v3#2675] [Q&A]
Hi everyone,

I'm trying to get the resource URL when uploading a file, however it doesn't return the location of the file:

Example response:
```
{
    "$metadata": {
        "httpStatusCode": 200,
        "extendedRequestId": "sdsdsdsdsddfdfdkfdfdfdfdf",
        "attempts": 1,
        "totalRetryDelay": 0
    },
    "ETag": "\"dsdsdsd36hgfhdgfhdfg\""
}
```

How I upload my file:
```
s3.send(
      new PutObjectCommand({
        Bucket: 'bucket,
        Key: 'key',
        Body: file.buffer,
        ACL: 'public-read',
      })
```

Why was the location key (resource URL) removed in version 3?

**Comment by jsefiani** [comment]
@trivikr Maybe you know?

**Reply by JanJakes** [reply]
Actually, this seems to be a bug introduced in `3.26.0`. After upgrade our uploads stopped working due to this (missing location), downgrading to `3.25.0` makes them work again.

**Reply by jsefiani** [reply]
Damn didn't know that, that's definitely a major bug...

Will report it at the end of the day, thanks for letting me know!

**Reply by ZimNovich** [reply]
Yes, the same story! The location key is missing.

**Reply by jsefiani** [reply]
Looks like downgrading, doesn't help 😢.

@ZimNovich Any luck?

**Reply by JanJakes** [reply]
Downgrading to `3.25.0` works for us. See also https://github.com/aws/aws-sdk-js-v3/issues/2694.

**Reply by wallind** [reply]
experiencing same problem trying to upgrade from `v2` -> `v3` for bundle size reduction benefits. Downgrading to `3.25.0` works for me as well as a temporary fix. Bumping to hopefully get this rectified.

**Reply by wallind** [reply]
for anyone watching this but not the related PR/Issue the PR fix was merged today https://github.com/aws/aws-sdk-js-v3/pull/2700. Not yet release but movement
